hledger-lib/Hledger/Data/Balancing.hs

```diff
@@ -146,6 +148,17 @@ transactionCheckBalanced BalancingOpts{commodity_styles_} t = errs
 isTransactionBalanced :: BalancingOpts -> Transaction -> Bool
 isTransactionBalanced bopts = null . transactionCheckBalanced bopts
 
+-- | Verify that any assertions in this transaction hold 
+-- when included in the larger journal.
+checkAssertions :: BalancingOpts -> Journal -> Transaction -> Either String Transaction
+checkAssertions bopts j t =
+  if (ignore_assertions_ bopts) || noassertions t then Right t else do
+    j' <- journalStyleAmounts j 
+    let newtxns = sortOn tdate (jtxns j' ++ [ t ])
+    fmap (\_ -> t) $ journalBalanceTransactions defbalancingopts j'{jtxns = newtxns}
```

Comment:
I guess that's a shorthand for
case ... of
  Right _ -> Right t
  Left e -> Left e
?